themes/classic/templates/_partials/microdata/product-jsonld.tpl

```diff
@@ -0,0 +1,83 @@
+ *}
+<script type="application/ld+json">
+  {
+    "@context": "https://schema.org/",
+    "@type": "Product",
+    "name": "{$product.name}",
+    "description": "{$page.meta.description|regex_replace:"/[\r\n]/" : " "}",
+    "category": "{$product.category_name}",
+    {if !empty($product.cover)}"image" :"{$product.cover.bySize.home_default.url}",{/if}
+    "sku": "{if $product.reference}{$product.reference}{else}{$product.id}{/if}",
+    "mpn": "{if $product.mpn}{$product.mpn}{elseif $product.reference}{$product.reference}{else}{$product.id}{/if}",
+    {if $product.ean13}"gtin13": "{$product.ean13}",{else if $product.upc}"gtin13": "{$product.upc}",{/if}
+    {if $product_manufacturer->name OR $shop.name}"brand": {
+      "@type": "Thing",
+      "name": "{if $product_manufacturer->name}{$product_manufacturer->name|escape:'html':'UTF-8'}{else}{$shop.name}{/if}"
+    },{/if}
+    {if isset($nbComments) && $nbComments && $ratings.avg}"aggregateRating": {
+      "@type": "AggregateRating",
+      "ratingValue": "{$ratings.avg|round:1|escape:'html':'UTF-8'}",
+      "reviewCount": "{$nbComments|escape:'html':'UTF-8'}"
+    },{/if}
+    {if isset($product.weight) && ($product.weight != 0)}
+      "weight": {
+        "@context": "https://schema.org",
+        "@type": "QuantitativeValue",
+        "value": "{$product.weight}",
+        "unitCode": "{$product.weight_unit}"
+      },{/if}
+      {if $product.show_price}
+          "offers": {
+            "@type": "Offer",
+            "priceCurrency": "{$currency.iso_code}",
+            "name": "{$product.name|strip_tags:false}",
+            "price": "{$product.price_amount}",
+            "url": "{$product.url}",
+            "priceValidUntil": "{($smarty.now + (int) (60*60*24*15))|date_format:"%Y-%m-%d"}",
+            {if $product.images|count > 0}
+              "image": {strip}[
+                {foreach from=$product.images item=p_img name="p_img_list"}
+                  "{$p_img.large.url}"{if not $smarty.foreach.p_img_list.last},{/if}
+                {/foreach}
+              ]{/strip},
+            {/if}
+            "sku": "{if $product.reference}{$product.reference}{else}{$product.id}{/if}",
+            "mpn": "{if $product.mpn}{$product.mpn}{elseif $product.reference}{$product.reference}{else}{$product.id}{/if}",
+            {if $product.ean13}"gtin13": "{$product.ean13}",{else if $product.upc}"gtin13": "0{$product.upc}",{/if}
+            {if $product.condition == 'new'}"itemCondition": "https://schema.org/NewCondition",{/if}
+            {if $product.show_condition > 0}
+              {if $product.condition == 'used'}"itemCondition": "https://schema.org/UsedCondition",{/if}
+              {if $product.condition == 'refurbished'}"itemCondition": "https://schema.org/RefurbishedCondition",{/if}
+            {/if}
+            "availability": "{if $product.quantity > 0 || $product.allow_oosp > 0}https://schema.org/InStock{else}https://schema.org/OutOfStock{/if}",
```

Comment:
@fdonnet I think you should use InStock only if the product is really in stock - `"availability":{if $product.quantity > 0} "http://schema.org/InStock"{else} "http://schema.org/OutOfStock"{/if},`

See: https://support.google.com/merchants/answer/6324448#zippy=%2Cproduct-inventory-ran-out%2Cproduct-is-available-for-sale%2Cvariants

Reply:
hi @Hlavtox, if i remember well you asked me to add $product.allow_oosp > 0 in my old PR... so really, it's like you prefer. Do I remove it and keep only the product quantity check ?

Reply:
@fdonnet Maybe, having second thoughts. I leave it up to you, your PR. 🤣 

@jf-viguier Wdyt? Should we put in stock also for backorder products? It would attract more customers, but I know google complains that my xml product data were different than the real data on the page.

Reply:
@fdonnet Aaaahaaa, originally it is done like this: `<link itemprop="availability" href="{$product.seo_availability}"/>`. I think we should keep it, wdyt? So `"availability": "{$product.seo_availability}",`

Reply:
@Hlavtox from what I know Google is planning to stop searching for differences between site and xml 🤔

Reply:
@kpodemski Hmm, nice point. Anyway, I think we should stick to what is provided by the core.

I proposed an issue with the availablity itself here: https://github.com/PrestaShop/PrestaShop/issues/23534

Reply:
@Hlavtox corrected. Put the responsability of the value in the define core smarty var ;) thx